src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -1370,6 +1370,9 @@ public InMemoryOutput downloadOutputs(RemoteAction action, RemoteActionResult re
         }
 
         if (isInMemoryOutputFile) {
+          if (shouldDownload(result, execPath, null)) {
+            remoteOutputChecker.omitDownload(inMemoryOutputPath); // Download into memory only; do not write to disk.
+          }
```

Comment:
I think it's better to not overthink this and unconditionally call `omitDownload`, even though it might lead to additional memory consumption. The fact that `shouldDownload` returned `false` does not imply that it will continue to return `false` for the same arguments in the future, because analysis and execution are interleaved and any time an additional target finishes analysis it may cause elements to be added to `pathsToDownload`. (I'm not sure that this can actually happen in practice for in-memory outputs, but even if it's true today, I'd rather not rely on it continuing to hold in the future.)

Reply:
I changed according to comment.